packages/components/src/components/wizard/wizard.tsx

```diff
@@ -1,42 +1,57 @@
-import PropTypes from 'prop-types';
+import PropTypes, { object } from 'prop-types';
 import React from 'react';
 import classNames from 'classnames';
-import Step from './wizard-step.jsx';
+import Step from './wizard-step';
+
+type TWizard = {
+    className?: string;
+    initial_step: number;
+    onStepChange: (prop: { [key: string]: number }) => void;
+    nav: React.ReactNode;
+    selected_step_ref: () => any | React.MutableRefObject<HTMLElement>;
+};
 
-const Wizard = ({ children, className, initial_step, onStepChange, nav, selected_step_ref }) => {
+const Wizard = ({
+    children = [],
+    className,
+    initial_step = 1,
+    onStepChange = () => undefined,
+    nav = null,
+    selected_step_ref,
+}: React.PropsWithChildren<TWizard>) => {
     const [active_step, setActiveStep] = React.useState(0);
 
+    const getSteps = React.useCallback(() => React.Children.toArray(children), [children]);
+
     React.useEffect(() => {
         const local_initial_step = initial_step - 1;
         const local_children = getSteps();
 
         if (local_initial_step && local_children[local_initial_step]) {
             setActiveStep(initial_step);
         }
-    }, [initial_step, getSteps]);
```

Comment:
@likhith-deriv  why did you remove `getSteps` ? :thinking:

Reply:
getSteps refers to React.useCallback(). Since the function ref doesn't change after component initialization, do we need to add it as a dependency for useEffect?

Reply:
@likhith-deriv you are correct about `getSteps` not changing but still anything that is being used inside a `useEffect` must be added to its dependency array.